fix(*): define missing custom properties from theme directory

cdransf opened the pull request:
## Description

`CSS-1084`

- [x] fixes missing custom properties by transferring them from `s2-foundations-redux`

![image](https://github.com/user-attachments/assets/27b2481a-841b-44b8-abad-187a4f3a9474)
![image](https://github.com/user-attachments/assets/644e8a14-48cd-40e9-aa33-832af4e45ca2)

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

marissahuysentruyt commented on components/actionbutton/index.css:
```diff
@@ -11,65 +11,18 @@
 	--spectrum-actionbutton-background-color-hover: var(--highcontrast-actionbutton-background-color-hover, var(--mod-actionbutton-background-color-hover, var(--spectrum-gray-100)));
 	--spectrum-actionbutton-background-color-down: var(--highcontrast-actionbutton-background-color-down, var(--mod-actionbutton-background-color-down, var(--spectrum-gray-200)));
 	--spectrum-actionbutton-background-color-focus: var(--highcontrast-actionbutton-background-color-focus, var(--mod-actionbutton-background-color-focus, var(--spectrum-gray-100)));
-	--spectrum-actionbutton-background-color-selected-disabled: var(--spectrum-transparent-black-100);
-
-	--spectrum-actionbutton-background-color-selected: var(--spectrum-neutral-background-color-selected-default);
-	--spectrum-actionbutton-background-color-selected-hover: var(--spectrum-neutral-background-color-selected-hover);
-	--spectrum-actionbutton-background-color-selected-down: var(--spectrum-neutral-background-color-selected-down);
-	--spectrum-actionbutton-background-color-selected-focus: var(--spectrum-neutral-background-color-selected-key-focus);
```
Looks like these were all just "Unused custom properties," which I get why we would remove. But are these the values we should be using? https://github.com/adobe/spectrum-css/blob/9c3a6253d4dd7208bcb9c2cab544c204e9cc9fcf/components/actionbutton/themes/spectrum-two.css#L20

How does that work is the `spectrum-two.css` in the foundations branch? It's late on a Friday so my brain might not be working hahaha

cdransf replied:
Ah, good catch! It looks like those are unused because we replaced fallback definitions for quite a few mods with more generic Spectrum tokens. Those custom properties had been used here:

```
--mod-actionbutton-background-color-default: var(--mod-actionbutton-background-color-default-selected, var(--spectrum-neutral-background-color-selected-default));
--mod-actionbutton-background-color-hover: var(--mod-actionbutton-background-color-hover-selected, var(--spectrum-neutral-background-color-selected-hover));
--mod-actionbutton-background-color-down: var(--mod-actionbutton-background-color-down-selected, var(--spectrum-neutral-background-color-selected-down));
--mod-actionbutton-background-color-focus: var(--mod-actionbutton-background-color-focus-selected, var(--spectrum-neutral-background-color-selected-key-focus));
```

I know — in a lot of cases — we'll create component-specific aliases for tokens, but the implementation on `spectrum-two` _isn't_ doing that. For example:

**s2-foundations-redux**

```css
--mod-actionbutton-background-color-default: var(--mod-actionbutton-background-color-default-selected, var(--spectrum-actionbutton-background-color-selected));
--mod-actionbutton-background-color-hover: var(--mod-actionbutton-background-color-hover-selected, var(--spectrum-actionbutton-background-color-selected-hover));
--mod-actionbutton-background-color-down: var(--mod-actionbutton-background-color-down-selected, var(--spectrum-actionbutton-background-color-selected-down));
--mod-actionbutton-background-color-focus: var(--mod-actionbutton-background-color-focus-selected, var(--spectrum-actionbutton-background-color-selected-focus));
--mod-actionbutton-background-color-disabled: var(--spectrum-actionbutton-background-color-selected-disabled);

--mod-actionbutton-border-color-default: transparent;
--mod-actionbutton-border-color-hover: transparent;
--mod-actionbutton-border-color-down: transparent;
--mod-actionbutton-border-color-focus: transparent;
--mod-actionbutton-border-color-disabled: transparent;

--mod-actionbutton-content-color-default: var(--mod-actionbutton-content-color-default-selected, var(--spectrum-actionbutton-content-color-selected));
--mod-actionbutton-content-color-hover: var(--mod-actionbutton-content-color-hover-selected, var(--spectrum-actionbutton-content-color-selected));
--mod-actionbutton-content-color-down: var(--mod-actionbutton-content-color-down-selected, var(--spectrum-actionbutton-content-color-selected));
--mod-actionbutton-content-color-focus: var(--mod-actionbutton-content-color-focus-selected, var(--spectrum-actionbutton-content-color-selected));
```

**spectrum-two**

```css
--mod-actionbutton-background-color-default: var(--mod-actionbutton-background-color-default-selected, var(--spectrum-neutral-background-color-selected-default));
--mod-actionbutton-background-color-hover: var(--mod-actionbutton-background-color-hover-selected, var(--spectrum-neutral-background-color-selected-hover));
--mod-actionbutton-background-color-down: var(--mod-actionbutton-background-color-down-selected, var(--spectrum-neutral-background-color-selected-down));
--mod-actionbutton-background-color-focus: var(--mod-actionbutton-background-color-focus-selected, var(--spectrum-neutral-background-color-selected-key-focus));

--mod-actionbutton-content-color-default: var(--mod-actionbutton-content-color-default-selected, var(--spectrum-gray-25));
--mod-actionbutton-content-color-hover: var(--mod-actionbutton-content-color-hover-selected, var(--spectrum-gray-25));
--mod-actionbutton-content-color-down: var(--mod-actionbutton-content-color-down-selected, var(--spectrum-gray-25));
--mod-actionbutton-content-color-focus: var(--mod-actionbutton-content-color-focus-selected, var(--spectrum-gray-25));
```

Redux used the aliasing pattern, but the spectrum-two branch doesn't, so I'm not sure which pattern or direction to take. 😅 Does the fact that they're mods factor in too?

marissahuysentruyt replied:
Happy Monday 👋  Just wanted to clarify this comment. It looks like in `s2-foundations-redux` that at least `--spectrum-actionbutton-background-color-selected-disabled` is in fact used: https://github.com/adobe/spectrum-css/blob/9c3a6253d4dd7208bcb9c2cab544c204e9cc9fcf/components/actionbutton/index.css#L136

I guess my question is the same, how are we reconciling what's going on in the foundations branch or main? I feel like I'm missing something 😵‍💫 maybe I'm thinking too hard, or not even looking in the right files.

cdransf replied:
Ahhh, that makes sense! I pushed up [08087d](https://github.com/adobe/spectrum-css/pull/3476/commits/08087dcf8e9c6cc1ac86c16d2c7b45243e09095f) which restores those and references that went missing from `s2-foundations-redux`.

marissahuysentruyt reviewed (commented):
Nice work on this! I think we just to rebuild the metadata and it'll be good to merge! I just wanted to double check with you on some of those action button "unused custom properties" (I left a comment). Otherwise, I didn't get any linter warnings/errors for each of the components ✅

cdransf commented on components/actionbutton/index.css:
```diff
@@ -11,65 +11,18 @@
 	--spectrum-actionbutton-border-color-disabled: var(--spectrum-disabled-border-color);
-
-	--spectrum-actionbutton-content-color-selected: var(--spectrum-gray-50);
+	--spectrum-actionbutton-content-color-disabled: var(--spectrum-disabled-content-color);
```
It is — it's not in `s2-foundations-redux` (the custom property isn't present in `index.css` or `spectrum-two.css` and it's not used in `index.css`). `--spectrum-disabled-content-color` is widely used though, so I'm comfortable with this alias (but I'm I don't have strong opinions either way).

rise-erpelding commented on components/dialog/index.css:
```diff
@@ -414,6 +414,6 @@
 @media (forced-colors: active) {
 	.spectrum-Dialog {
-		border: 1px solid var(--highcontrast-standard-dialog-border-color, rgba(var(--spectrum-black-rgb), var(--spectrum-overlay-opacity)));
+		border: 1px solid rgba(var(--spectrum-black-rgb), var(--spectrum-overlay-opacity));
```
Judging by what Marissa had done in her dialog migration PR #2860, this is probably the right call, but I'm not able to verify what it looks like because Dialog is looking like this for me:
![image](https://github.com/user-attachments/assets/1f6f4220-e145-423a-8c81-1cf9dcd30d45)
Is anyone else having the same experience? If so, maybe we can make a follow-up card to look into this issue since it seems unrelated to the current fixes in this PR (if one doesn't already exist)?

Unless @marissahuysentruyt remembers the migration well enough and can spot something missing that would fix it here.

cdransf replied:
Got it! I'm seeing the same but had gone ahead and lifted it from the S2 theme file on the redux branch. I'll wait for @marissahuysentruyt to chime in,

marissahuysentruyt replied:
@rise-erpelding @cdransf I do have a fix for this in the takeover dialog PR! https://github.com/adobe/spectrum-css/pull/3347/files#r1878562052 `layout: fullscreen` 

And yes, Cory and I talked about the removal of that highcontrast variable. I think it makes sense. We never defined it- I added it as if it were a mod, but that was the incorrect usage.

rise-erpelding commented on components/actionbutton/index.css:
```diff
@@ -144,15 +102,17 @@
-		--mod-actionbutton-background-color-down: var(--mod-actionbutton-background-color-down-selected, var(--spectrum-neutral-background-color-selected-down));
-		--mod-actionbutton-background-color-focus: var(--mod-actionbutton-background-color-focus-selected, var(--spectrum-neutral-background-color-selected-key-focus));
-
-		--mod-actionbutton-content-color-default: var(--mod-actionbutton-content-color-default-selected, var(--spectrum-gray-25));
```
Do we still need this (and maybe the hover, down, focus colors here)? I'm seeing selected content color coming through like this now, looking black:
![image](https://github.com/user-attachments/assets/d6354f44-e7bc-49ef-83d4-91baeb5682b7)

cdransf replied:
Ohh nice catch! The first block of mods had been restored, but not `--mod-actionbutton-content-color-default`. Restoring that fixes the selected button state:

<img width="302" alt="image" src="https://github.com/user-attachments/assets/7e2312da-7ed7-4c29-95b8-1a4f8f280997" />

rise-erpelding reviewed (changes requested):
Wow, this ended up being quite the collection of components, especially with actionbutton and dialog in the mix! It looks like you took care of all the no undefined custom props warnings, so that's great!

I think there are a few remaining oddities particularly with actionbutton, like the selected state content color (and also the default/disabled state border, which I didn't comment on within the code because I wasn't able to figure out if that was because of a missing/undefined custom prop or something else). Hopefully we can resolve them here, or if not, in the S2 migration. I'd also noticed that when I was running `yarn build` on this branch that there were some lingering metadata changes that needed to be committed, but that may or may not still be an issue if your branch has been updated since then.

cdransf commented:
> Wow, this ended up being quite the collection of components, especially with actionbutton and dialog in the mix! It looks like you took care of all the no undefined custom props warnings, so that's great!
> 
> I think there are a few remaining oddities particularly with actionbutton, like the selected state content color (and also the default/disabled state border, which I didn't comment on within the code because I wasn't able to figure out if that was because of a missing/undefined custom prop or something else). Hopefully we can resolve them here, or if not, in the S2 migration. I'd also noticed that when I was running `yarn build` on this branch that there were some lingering metadata changes that needed to be committed, but that may or may not still be an issue if your branch has been updated since then.

I left a few comments, fixed the `actionbutton` selected state and updated the metadata ✨

cdransf commented on components/actionbutton/index.css:
```diff
@@ -144,15 +102,19 @@
 		--mod-actionbutton-content-color-default: var(--mod-actionbutton-content-color-default-selected, var(--spectrum-gray-25));
-		--mod-actionbutton-content-color-hover: var(--mod-actionbutton-content-color-hover-selected, var(--spectrum-gray-25));
-		--mod-actionbutton-content-color-down: var(--mod-actionbutton-content-color-down-selected, var(--spectrum-gray-25));
-		--mod-actionbutton-content-color-focus: var(--mod-actionbutton-content-color-focus-selected, var(--spectrum-gray-25));
```
Thank you! Just fixed that,

cdransf commented on components/clearbutton/index.css:
```diff
@@ -22,6 +21,11 @@
 	--spectrum-clear-button-icon-color-down: var(--spectrum-neutral-content-color-down);
 	--spectrum-clear-button-icon-color-key-focus: var(--spectrum-neutral-content-color-key-focus);
 
+	--spectrum-clear-button-background-color: transparent;
+	--spectrum-clear-button-background-color-hover: transparent;
+	--spectrum-clear-button-background-color-down: transparent;
+	--spectrum-clear-button-background-color-key-focus: transparent;
+
```
Ah yes! I believe so. Just added those as well. ✨

rise-erpelding reviewed (dismissed):
Looks good to me! I know @marissahuysentruyt was planning to take another look at this, I'd be curious to know if her eagle eyes can spot anything here.

Documenting that I still think actionbutton is missing the basebutton styles and that's what's causing the two-colored outset border style, but we weren't able to get those base button styles to work on action button, so there may need to be a follow-up card for that (or we'll address it in the migration).
![image](https://github.com/user-attachments/assets/e795b782-d45b-45c0-9d58-5d8b3c0626ae)

marissahuysentruyt commented on components/actionbutton/index.css:
```diff
@@ -132,7 +92,6 @@
 		--mod-actionbutton-border-color-focus: var(--spectrum-transparent-white-600);
 
 		--spectrum-actionbutton-background-color-disabled: transparent;
-		--spectrum-actionbutton-background-color-selected-disabled: var(--spectrum-transparent-white-200);
```
Here's what I was seeing before, without the background-color-selected-disabled and content-color-selected variables:

<img width="491" alt="Screenshot 2025-01-08 at 10 09 12 AM" src="https://github.com/user-attachments/assets/6faad1fc-07d3-45e8-b810-6d94cab8747e" />

But I think we want something more like this, with the background-color-selected-disabled? I'm trying not to just fix styles and save that for the migration (hence why the content color is still incorrect), but I think the background-color-selected-disabled should be defined here.

<img width="421" alt="Screenshot 2025-01-08 at 10 29 01 AM" src="https://github.com/user-attachments/assets/53b1c967-dc8f-4fd5-8944-a7d7860eaa82" />

cdransf replied:
Just updated this!

marissahuysentruyt reviewed (commented):
I believe I found one or two more variables that _should_ be defined here- let me know what you think!

marissahuysentruyt commented on components/actionbutton/index.css:
```diff
@@ -114,7 +74,6 @@
 		--mod-actionbutton-border-color-focus: var(--spectrum-transparent-black-600);
 
 		--spectrum-actionbutton-background-color-disabled: transparent;
-		--spectrum-actionbutton-background-color-selected-disabled: var(--spectrum-transparent-black-200);
```
Do we need this line still? It looks like the static white has both a disabled and a selected-disabled custom property, so maybe we still need both properties here. What do you think?

marissahuysentruyt replied:
@cdransf I thought we had added this back in for the static black, but we must have reverted it. That way both static white & static black has their own, specific `--spectrum-actionbutton-background-color-selected-disabled` value.

rise-erpelding reviewed (commented):
Wanted to drop a quick comment since I know this is still in review!

Since actionbutton particularly has been causing problems, I dug into this a bit more to try to figure out what's going on with it. Comparing how it's changed from `s2-foundations-redux`, I noticed that the styles that previously applied with the import of `commons/basebutton.css` are now hardcoded into the component, like `cursor: pointer;`, `user-select: none;` etc.

We lost the `border-style: solid;` in commons/basebutton.css in this [commit](https://github.com/adobe/spectrum-css/commit/cc3b7d7645b9e4cf500e95fa6e07d37f069cf92b#diff-715275ed27a56c42d8cf769b25514cbd54a4fd44bf5349430f0acd9ba1a268f2) so it didn't make it in when the commons styles were pasted in in this later [commit](https://github.com/adobe/spectrum-css/pull/2352/commits/9444cdbb3a6d8c2d880b53c691c0c067e04e7e1f#diff-b86e2a5a1675710bcd76aa3ceec37bdabcfcbf4082593cdc602f41712a6165a0). That was the style we needed for the button outlines and importing basebutton in this branch wasn't doing us any good because that style isn't in basebutton anymore.

Anyway, a couple of takeaways from that knowledge:
1. We should probably paste that `border-style: solid;`, probably below this section:

```
/* Remove the inheritance of text transform on button in Edge, Firefox, and IE. */
	text-transform: none;

	/* stylelint-disable-next-line property-no-vendor-prefix -- Correct the inability to style clickable types in iOS and Safari (normalize). */
	-webkit-appearance: button;
```

2. If things are getting really hairy with fixes, it might help to refer back to the way the code looked on the redux branch and how it changed between there and here. We know we have to add in any missing themes custom properties and ideally that would be it, but hopefully we can uncover any issues outside of that by looking back. 🤞

cdransf commented:
> Wanted to drop a quick comment since I know this is still in review!
> 
> Since actionbutton particularly has been causing problems, I dug into this a bit more to try to figure out what's going on with it. Comparing how it's changed from `s2-foundations-redux`, I noticed that the styles that previously applied with the import of `commons/basebutton.css` are now hardcoded into the component, like `cursor: pointer;`, `user-select: none;` etc.
> 
> We lost the `border-style: solid;` in commons/basebutton.css in this [commit](https://github.com/adobe/spectrum-css/commit/cc3b7d7645b9e4cf500e95fa6e07d37f069cf92b#diff-715275ed27a56c42d8cf769b25514cbd54a4fd44bf5349430f0acd9ba1a268f2) so it didn't make it in when the commons styles were pasted in in this later [commit](https://github.com/adobe/spectrum-css/pull/2352/commits/9444cdbb3a6d8c2d880b53c691c0c067e04e7e1f#diff-b86e2a5a1675710bcd76aa3ceec37bdabcfcbf4082593cdc602f41712a6165a0). That was the style we needed for the button outlines and importing basebutton in this branch wasn't doing us any good because that style isn't in basebutton anymore.
> 
> Anyway, a couple of takeaways from that knowledge:
> 
> 1. We should probably paste that `border-style: solid;`, probably below this section:
> 
> ```
> /* Remove the inheritance of text transform on button in Edge, Firefox, and IE. */
> 	text-transform: none;
> 
> 	/* stylelint-disable-next-line property-no-vendor-prefix -- Correct the inability to style clickable types in iOS and Safari (normalize). */
> 	-webkit-appearance: button;
> ```
> 
> 2. If things are getting really hairy with fixes, it might help to refer back to the way the code looked on the redux branch and how it changed between there and here. We know we have to add in any missing themes custom properties and ideally that would be it, but hopefully we can uncover any issues outside of that by looking back. 🤞

I just reverted 

> Wanted to drop a quick comment since I know this is still in review!
> 
> Since actionbutton particularly has been causing problems, I dug into this a bit more to try to figure out what's going on with it. Comparing how it's changed from `s2-foundations-redux`, I noticed that the styles that previously applied with the import of `commons/basebutton.css` are now hardcoded into the component, like `cursor: pointer;`, `user-select: none;` etc.
> 
> We lost the `border-style: solid;` in commons/basebutton.css in this [commit](https://github.com/adobe/spectrum-css/commit/cc3b7d7645b9e4cf500e95fa6e07d37f069cf92b#diff-715275ed27a56c42d8cf769b25514cbd54a4fd44bf5349430f0acd9ba1a268f2) so it didn't make it in when the commons styles were pasted in in this later [commit](https://github.com/adobe/spectrum-css/pull/2352/commits/9444cdbb3a6d8c2d880b53c691c0c067e04e7e1f#diff-b86e2a5a1675710bcd76aa3ceec37bdabcfcbf4082593cdc602f41712a6165a0). That was the style we needed for the button outlines and importing basebutton in this branch wasn't doing us any good because that style isn't in basebutton anymore.
> 
> Anyway, a couple of takeaways from that knowledge:
> 
> 1. We should probably paste that `border-style: solid;`, probably below this section:
> 
> ```
> /* Remove the inheritance of text transform on button in Edge, Firefox, and IE. */
> 	text-transform: none;
> 
> 	/* stylelint-disable-next-line property-no-vendor-prefix -- Correct the inability to style clickable types in iOS and Safari (normalize). */
> 	-webkit-appearance: button;
> ```
> 
> 2. If things are getting really hairy with fixes, it might help to refer back to the way the code looked on the redux branch and how it changed between there and here. We know we have to add in any missing themes custom properties and ideally that would be it, but hopefully we can uncover any issues outside of that by looking back. 🤞

Alrighty! I just added that missing rule. I had tried reintroducing the action button updates by stepping through the changes again, but things still weren't quite right.

rise-erpelding commented on components/actionbutton/index.css:
```diff
@@ -144,15 +105,16 @@
-		--mod-actionbutton-content-color-hover: var(--mod-actionbutton-content-color-hover-selected, var(--spectrum-gray-25));
-		--mod-actionbutton-content-color-down: var(--mod-actionbutton-content-color-down-selected, var(--spectrum-gray-25));
-		--mod-actionbutton-content-color-focus: var(--mod-actionbutton-content-color-focus-selected, var(--spectrum-gray-25));
+		--mod-actionbutton-content-color-default: var(--mod-actionbutton-content-color-default-selected, var(--spectrum-actionbutton-content-color-selected));
```
Are we missing this [chunk](https://github.com/adobe/spectrum-css/blob/s2-foundations-redux/components/actionbutton/index.css#L138-L142) of border-color mods that s2-foundations has about here?

cdransf replied:
Just added those back — thank you! ✨

rise-erpelding commented on components/actionbutton/index.css:
There's this selector in [s2 foundations](https://github.com/adobe/spectrum-css/blob/s2-foundations-redux/components/actionbutton/index.css#L68) that sets the border color to transparent that looks like its missing here, after `--spectrum-logical-rotation`. You have this selector here but I think would maybe need to add the border-color?

```css
&.spectrum-ActionButton--quiet {
		--spectrum-actionbutton-border-color: transparent;
	}
```

Except that I don't think that works because now we're not setting `--spectrum-actionbutton-border-color` anywhere, so maybe try something like
```css
                 --mod-actionbutton-border-color-default: transparent;
		--mod-actionbutton-border-color-hover: transparent;
		--mod-actionbutton-border-color-down: transparent;
		--mod-actionbutton-border-color-focus: transparent;
		--mod-actionbutton-border-color-disabled: transparent;
```

----------

We're also missing [setting](https://github.com/adobe/spectrum-css/blob/s2-foundations-redux/components/actionbutton/index.css#L104C3-L104C96) the static-black-focus-indicator-color:

```css
--mod-actionbutton-focus-indicator-color: var(--spectrum-static-black-focus-indicator-color);
```

And [static white](https://github.com/adobe/spectrum-css/blob/s2-foundations-redux/components/actionbutton/index.css#L128C3-L128C96):
```css
--mod-actionbutton-focus-indicator-color: var(--spectrum-static-white-focus-indicator-color);
```

cdransf replied:
I believe the `spectrum-action-border-color` change(s) should be handled in this commit: [fef056](https://github.com/adobe/spectrum-css/pull/3476/commits/fef056c39b9d42d1da83b80a5d76d4091c67aa0e). I stepped through the states on this branch/PR and matched them to a recent deploy with the previous, working state of the button(s).

![image](https://github.com/user-attachments/assets/e101899f-b78e-4c7e-a9a2-2e04cec02efa)
![image](https://github.com/user-attachments/assets/fd718dde-fc9c-4d00-95e9-a706e044dc67)

rise-erpelding reviewed (changes requested):
Noting a couple of diffs I saw between this branch and s2-foundations that are causing some regressions!

rise-erpelding commented:
Hey! I think we've come a long way on action button and while we're still getting unintended borders on several quiet variants, I think a lot of this is due to some of the structural differences between the branches, which isn't really what we intended to fix here, and is proving to be a lot more difficult than originally anticipated. Overall, it looks fine and there are no glaring contrast or background issues that I'm seeing. If there are no other improvements that you or @marissahuysentruyt see to make on action button, I think that one's good to go.

When scanning the rest of the components fixed here, I did still notice some possible inconsistencies with button that I'd love some opinions on:
![image](https://github.com/user-attachments/assets/b8a9a17a-be7a-4dee-9c70-18cedbe69ef2)
In this image, I think the second set of accent/negative buttons is supposed to be the outline variant? But also, I think button has already been migrated to S2 in this branch and it looks as if maybe those variants don't exist anymore. Could you confirm? It seems reasonable to think that we might need to remove test cases or maybe make some adjustments to the template after all the updates to `spectrum-two`, and if that's the case, it'd be ok to make another card for that.

As far as I can tell I think the other components look good!

rise-erpelding commented on tokens/dist/index.css:
Is this file deletion intentional?

cdransf replied:
It wasn't! Should be restored now. ✨

marissahuysentruyt reviewed (commented):
Running `yarn linter` against each component doesn't return any errors 🥳 With the project running, I don't see anything that's absolutely broken (wrong styles - yes, but nothing totally broken or unusable). I was a little concerned at the number of mods that look like they've been deleted in action button, but you mentioned in standup that they seemed self-referential, so maybe it's ok for the `spectrum-two` branch. 

Like @rise-erpelding said, there's lots of work that probably could be done with action button. We'll have to readdress those things when the action button migration gets restarted. I also see the missing outlines in the test cases for outline buttons (she left that screenshot).

marissahuysentruyt commented on components/actionbutton/index.css:
```diff
@@ -69,7 +29,9 @@
 
-	--spectrum-actionbutton-content-color-selected: var(--spectrum-gray-50);
+	--spectrum-actionbutton-content-color-selected: var(--spectrum-gray-25);
+	--spectrum-actionbutton-content-color-disabled: var(--mod-disabled-content-color, var(--spectrum-disabled-content-color));
+	--spectrum-actionbutton-background-color-disabled: var(--mod-disabled-background-color, var(--spectrum-disabled-background-color)); 
```
For these 2 new mods (mod-disabeld-content-color and mod-disabled-background-color), what do you think of naming them more in the format we have for the other mods? 

```
--mod-actionbutton-content-color-disabled
--mod-actionbutton-background-color-disabled
```

Just a thought to match the other properties.

marissahuysentruyt reviewed (approved):
Thanks for all your hard work on this!

rise-erpelding reviewed (approved):
I think this is good to go!

Re-summarizing my notes:
* Action button has several instances of visible borders where they should be transparent, but this doesn't seem to be because of missing custom properties so hopefully this can be taken care of in the migration.
* Button - some follow up should be done to remove unneeded outline variants from tests since it's already been migrated and those variants don't exist anymore.

Clear button, Avatar, Dialog, and Infield button look good! Thanks @cdransf!
